SC_redesign/Crop_and_Soil/soil/carbon_cycling/residue_partition.py

```diff
@@ -17,6 +18,124 @@ class ResiduePartition:
+                    self.data.decomposition_temperature_effect,
+                    layer.plant_structural_carbon_amount)
+
+            layer.weighted_residue_dry_matter_lignin_fraction = \
+                self._determine_weighted_residue_dry_matter_lignin_fraction(layer.soil_dry_matter_residue_amount,
+                                                                            crop.root_biomass)
+
+            layer.soil_residue_lignin_fraction = self._determine_soil_residue_lignin_fraction(
+                layer.weighted_residue_dry_matter_lignin_fraction,
+                rainfall
+            )
+
+            layer.soil_lignin_to_nitrogen_fraction = self._determine_soil_lignin_to_nitrogen_fraction(
+                self.data.plant_residue_metabolic_fraction,
+                layer.weighted_residue_dry_matter_lignin_fraction,
+                layer.soil_residue_lignin_fraction,
+            )
+
+            layer.soil_residue_metabolic_fraction = self._determine_soil_residue_metabolic_fraction(
+                layer.soil_lignin_to_nitrogen_fraction
+            )
+
+            layer.soil_metabolic_active_carbon_usage = self._determine_soil_metabolic_to_active_carbon_amount(
+                layer.decomposition_moisture_effect,
+                self.data.decomposition_temperature_effect,
+                layer.soil_metabolic_carbon_amount,
+            )
+
+            layer.soil_metabolic_carbon_amount = self._determine_soil_metabolic_carbon_amount(
+                layer.soil_metabolic_carbon_amount,
+                layer.plant_metabolic_to_soil_carbon_amount,
+                crop.root_biomass,
+                layer.soil_residue_metabolic_fraction,
+                layer.soil_metabolic_active_carbon_usage
+            )
+
+            layer.soil_structural_active_carbon_usage = self._determine_soil_structural_to_slow_active_carbon_amount(
+                layer.decomposition_moisture_effect,
+                self.data.decomposition_temperature_effect,
+                layer.soil_structural_carbon_amount
+            )
+
+            layer.soil_structural_slow_carbon_usage = self._determine_soil_structural_to_slow_active_carbon_amount(
+                layer.decomposition_moisture_effect,
+                self.data.decomposition_temperature_effect,
+                layer.soil_structural_carbon_amount
+            )
+
+            layer.soil_structural_carbon_amount = self._determine_soil_structural_carbon_amount(
+                layer.soil_residue_metabolic_fraction,
+                layer.structural_carbon_transfer_amount,
+                layer.soil_structural_active_carbon_usage,
+                layer.soil_structural_slow_carbon_usage,
+                crop.root_biomass,
+                layer.soil_structural_carbon_amount
+            )
+
     @staticmethod
     def _determine_plant_residue_lignin_composition(plant_residue_lignin_composition: float,
                                                     rainfall: float) -> None:
```

Comment:
This method should return a `float`, right?

```suggestion
    def _determine_plant_residue_lignin_composition(plant_residue_lignin_composition: float,
                                                    rainfall: float) -> float:
```